src/js/base/runtime.js

```diff
@@ -3072,7 +3072,7 @@ function (Namespace, jsnums, codePoint, seedrandom, util) {
         stackFrame = $ar.args[2];
         $fun_ans = $ar.vars[0];
       }
-      if (--thisRuntime.GAS <= 0 || --thisRuntime.RUNGAS <= 0) {
+      if (thisRuntime.bounceAllowed && (--thisRuntime.GAS <= 0 || --thisRuntime.RUNGAS <= 0)) {
```

Comment:
Oooh this is frustrating to see the copy/pasted pattern.  Hopefully this is simple enough that if we put it in a helper it would be inlined by the major engines.